### ISSUE TITLE
Huge Memory usage and CPU spikes

### ISSUE THREAD

**Issue by dhilgarth** [gitextensions/gitextensions#3695]
**Do you want to request a *feature* or report a *bug*?**
Report a bug.

**What is the current behavior?**
GitExtensions uses a huge amount of memory for no apparent reason. Also, it causes periodic CPU usage spikes:
![image](https://cloud.githubusercontent.com/assets/1180105/25921732/d84d8be0-35d6-11e7-908d-229107a8638f.png)  
(this is a quad core machine with Hyper Threading and 32 GB memory)

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**
This happens when GitExtensions is running for extended periods of time. It might be connected to the number of tracked and untracked files in the repository as there are quite a lot in the repository.

**What is the expected behavior?**
No CPU spikes, no GBs of memory usage

**Which versions of GIT and which versions of Windows are affected by this issue? Did this work in previous versions of our tool?**
GitExtensions version 2.49.03, Windows 10 64bit. I have noticed this for some time now, also with previous versions if I remember correctly.

**Comment by vbjay** [comment]
Do you have the option that shows the working directory changes in the revision graph turned on?  Do you have background fetch set up?  Try using procmon or procexp to watch it and see what it is doing at the time.

**Comment by dhilgarth** [comment]
Thanks for the hints with procmon and procexp. Will use this the next time it uses so much memory. I had to kill it for now.

* Show working directory changes: No
* Background fetch: Not sure what that setting is.
Can you get that info from the screenshot?

![image](https://cloud.githubusercontent.com/assets/1180105/25923620/a48a44ee-35de-11e7-81a4-353862e1828f.png)

**Comment by vbjay** [comment]
It's in plugins.

**Comment by dhilgarth** [comment]
It's disabled.

**Comment by vcpp** [comment]
@dhilgarth, do you have sensitive content in your Git Extensions process?
If not or if you don't care a lot about privacy, you could save full process memory dump next time you see the issue and share it somehow to let me inspect the memory leak.

Previously I've experienced troubles with TeamCity integration: #3646. I didn't fix it yet, since this happened only once during TeamCity server maintenance and so it doesn't seem to be critical for me.

**Comment by dhilgarth** [comment]
Do you really want to inspect a 17GB memory dump?

![image](https://user-images.githubusercontent.com/1180105/26966854-15dcf19e-4cfc-11e7-9d3a-78673aa195c5.png)

**Comment by dhilgarth** [comment]
But the hint with the TeamCity integration was great. Disabling it stopped the CPU usage. And most likely, the memory bloat as well.

**Comment by vcpp** [comment]
> Do you really want to inspect a 17GB memory dump?

The question is whether to inspect or not. Size of the dump doesn't really matter, it impacts hardware only, since I'm not going to read individual bytes, there is no linear dependency for human :-)

However, if disabling TeamCity integration solves the memory consumption issue, it's most likely a duplicate or something related to #3646. I suspect careless using of Tasks to be the root of evil. There is no point in wasting time on this investigation before fixing #3646.

**Comment by dhilgarth** [comment]
I agree

**Comment by drewnoakes** [comment]
It seems like this is a duplicate of #3646 so I'll close this and update the title to mention TeamCity. Re-open if you disagree.
